### ISSUE TITLE
Motion prediction along the z axis

### ISSUE THREAD

**Issue by toyot-li** [TRI-ML/PF-Track#10]
@ziqipang Hello, thanks for your great work.

I find only the first two dimensions are included in the loss calculation of prediction, and I noticed that you have set `find_unused_parameters=False`. Where does the third dimension of the motion regression head receive the gradient? Am I missing anything important? Thanks.

**Comment by ziqipang** [comment]
@zen-d Thanks for your question! I only use the motion on the BEV for updating reference points (https://github.com/TRI-ML/PF-Track/blob/a254761a30bb186e38262eb82fca509065cbad42/projects/tracking_plugin/models/trackers/spatial_temporal_reason.py#L261). However, what you mentioned is correct and interesting, and I haven't paid attention to it before. Somehow the training runs correctly even though the z-axis is not used.

**Comment by ziqipang** [comment]
@zen-d Yeah, I agree with you. I am also surprised after noticing what you mentioned to me. I tend to think this is a behavior of DDP and not so familiar with it. Perhaps you could have a minimal example of exploring this and ask some experts on the Pytorch forum.
